File: src/main/java/mil/nga/tiff/field/GeoTiffTag.java
package mil.nga.tiff.field;

/**
 * GEO TIFF tags
 */
public enum GeoTiffTag implements FieldTagType {
    ModelPixelScale(33550, false),
    ModelTiepoint(33922, false),
    ModelTransformation(34264, false),
    GeoKeyDirectory(34735, false),
    GeoDoubleParams(34736, false),
    GeoAsciiParams(34737, false);

    /**
     * Tag id
     */
    private final int id;

    /**
     * True if an array type
     */
    private final boolean array;

    /**
     * Constructor
     *
     * @param id    tag id
     * @param array true if an array type
     */
    GeoTiffTag(int id, boolean array) {
        this.id = id;
        this.array = array;
    }

    /**
     * Is this field an array type
     *
     * @return true if array type
     */
    @Override
    public boolean isArray() {
        return array;
    }

    /**
     * Get the tag id
     *
     * @return tag id
     */
    @Override
    public int getId() {
        return id;
    }

}

File: src/main/java/mil/nga/tiff/field/tag/GeoTiffTag.kt
package mil.nga.tiff.field.tag

/**
 * GEO TIFF tags
 *
 * @param id      tag id
 * @param isArray true if an array type
 */
enum class GeoTiffTag(override val id: Int, override val isArray: Boolean) : FieldTagType {
    ModelPixelScale(33550, false),
    ModelTiepoint(33922, false),
    ModelTransformation(34264, false),
    GeoKeyDirectory(34735, false),
    GeoDoubleParams(34736, false),
    GeoAsciiParams(34737, false)
}
